liquidprompt

```diff
@@ -1397,9 +1459,63 @@ _lp_kubernetes_context() {
 _lp_kubernetes_context_color() {
     _lp_kubernetes_context || return "$?"
 
-    lp_kubernetes_context_color="[${LP_MARK_KUBECONTEXT}${LP_COLOR_KUBECONTEXT}${lp_kubernetes_context}${lp_kubernetes_namespace+:}${lp_kubernetes_namespace-}${NO_COL}]"
+    lp_kubernetes_context_color="${LP_MARK_KUBECONTEXT}${LP_COLOR_KUBECONTEXT}${lp_kubernetes_context}${lp_kubernetes_namespace+:}${lp_kubernetes_namespace-}${NO_COL}"
 }
 
+
+_lp_cmake() {
+    ((  LP_ENABLE_CMAKE  )) || return 2
+    [[ -f CMakeCache.txt ]] || return 1
+
+    _lp_grep_fields "CMakeCache.txt" "=" "CMAKE_C_COMPILER:FILEPATH"  "CMAKE_CXX_COMPILER:FILEPATH" "CMAKE_GENERATOR:INTERNAL" "CMAKE_BUILD_TYPE:STRING"
+
+    local cmake_c_compiler=${lp_grep_fields[_LP_FIRST_INDEX+0]}
```

Comment:
Need to protect all these if `set -u` is set.
```suggestion
    local cmake_c_compiler=${lp_grep_fields[_LP_FIRST_INDEX+0]-}
```
It looks like the rest of these functions will work "correctly" if these return nothing, though not optimized since it will hash and then color the empty string.